src/SFA.DAS.Payments.MatchedLearner.Data/SqlAzureIdentityTokenProvider.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+
+namespace SFA.DAS.Payments.MatchedLearner.Data
+{
+    public interface ISqlAzureIdentityTokenProvider
+    {
+        Task<string> GetAccessTokenAsync(CancellationToken cancellationToken = default);
+        string GetAccessToken();
+    }
+
+    public class SqlAzureIdentityTokenProvider : ISqlAzureIdentityTokenProvider
+    {
+        private readonly ILogger<SqlAzureIdentityTokenProvider> _logger;
+        private static readonly Microsoft.Azure.Services.AppAuthentication.AzureServiceTokenProvider AzureServiceTokenProvider = new Microsoft.Azure.Services.AppAuthentication.AzureServiceTokenProvider();
```

Comment:
Few things:

- fully qualified type name used but not needed
- variable marked as static but access is private and class is non static, not sure if intended
- doesn't follow underscore convention
- if will be changed to non static, should move assignment of variable to constructor